packages/carbon-graphs/src/js/helpers/axis.js

```diff
@@ -922,7 +928,18 @@ const createAxes = (axis, scale, config, canvasSVG) => {
       .call(resetD3FontSize);
   }
 };
-
+const calculateRotateAngleBasedOnWidth = (config) => {
+  const containerWidth = config.canvasWidth;
+  // Define thresholds for rotation based on container width
+  const thresholdSmall = 400;
+  const thresholdMedium = 800;
+  if (containerWidth < thresholdSmall) {
+    return -45; // rotate by -45 degrees for small screens
+  } if (containerWidth < thresholdMedium) {
+    return -30; // rotate by -30 degrees for medium screens
+  }
+  return -15; // rotate by -15 degrees for larger screens
+};
```

Comment:
I believe the solution we agreed upon was to make the prop an integer and let consumers choose the angle as per their own requirements. What if a consumer wants to rotate it by 60º? Or what if they would like to rotate it in the other direction like positive 45º? By hardcoding the values and specifying them per screen size, we risk interpreting their data in ways they do not want it to.

Reply:
resolved in https://github.com/cerner/terra-graphs/pull/322/commits/3a9d13ef45d724a7f346bd2672c4c13ecb22d5c2

Reply:
@sdadn we had meetings after that meeting where we decided that we should restrict the available values to the default horizontal or -45º

cc: @mjpalazzo